bots/communication.py

```diff
@@ -153,7 +153,7 @@ def run(self):
                         # ~ #max_nr_retry : get this from channel. should be integer, but only textfields where left. so might be ''/None->use 0
                         # ~ max_nr_retry = int(self.channeldict['rsrv1']) if self.channeldict['rsrv1'] else 0
                         # ~ if max_nr_retry:
-                            # ~ domain = (self.channeldict['idchannel'] + u'_failure')[:35]
+                            # ~ domain = (self.channeldict['idchannel'] + '_failure')[:35]
```

Comment:
im not sure about why this section is commented out but updated the code anyway in case it goes back in

Reply:
The else block should go before finally, then un-comment it.
Logic here is to only report when max_nr_retry errors in a row. As soon as there is a successful connection the counter should be reset to zero

Reply:
I also like the idea of max_nr_connect_tries; this will help with transient timeouts but not if a server is really down for some period.

Reply:
fixed this, see commit https://github.com/eppye-bots/bots3.3/pull/2/commits/f2eec06213919b4b16d8002e1290b76de4f5554b